src/module/mod.rs

```diff
@@ -128,9 +129,21 @@ impl Module {
         ModuleConfig::new().parse(wasm)
     }
 
+    /// Merge a module from the in-memory wasm buffer with the default configuration
+    pub fn merge_buffer(&mut self, wasm: &[u8]) -> Result<()> {
+        return Self::parse_in(wasm, &ModuleConfig::default(), self);
+    }
+
     fn parse(wasm: &[u8], config: &ModuleConfig) -> Result<Module> {
         let mut ret = Module::default();
         ret.config = config.clone();
+        Self::parse_in(wasm, config, &mut ret)?;
+        return Ok(ret);
+    }
+    fn parse_in(wasm: &[u8], config: &ModuleConfig, ret: &mut Module) -> Result<()> {
```

Comment:
Instead of having `ret` as the last arg, perhaps just make it the first `self` arg?

Reply:
Also, instead of taking a new `config`, we should probably use the same configuration, `ret.config` I think.